### ISSUE TITLE
Laravel queue:work and $mqqt->disconnect()

### ISSUE THREAD

**Issue by stijn1989** [php-mqtt/laravel-client#18]
This is a huge tip for the users of this library. It can safe some frustrated hours.

I have an API. When an endpoint is called, it starts a job on the queue. In this job, a MQTT message is published. The code below is the code that runs in the job.

```php
$topic = sprintf("Vennens/MVA/%d/Silo/%d/Alternatieven", $this->plc, $this->silo);
$mqtt = MQTT::connection();
$mqtt->publish($topic, json_encode([
    'alternatief_1' => $this->alternatief_1,
    'alternatief_2' => $this->alternatief_2,
    'alternatief_3' => $this->alternatief_3,
    'alternatief_4' => $this->alternatief_4,
]), 2);

$mqtt->loop(true, true, 5);
$mqtt->disconnect();
```

The first job process goes great. But when I do a second request. The job failes. You get an exception that there is no connection.

![image](https://user-images.githubusercontent.com/6877485/148641887-9686f519-c97b-44ca-b307-6e2cea824c05.png)

The problem is that the Laravel queue worker keeps it's app state. In the MQTT class, it stores the connections in an array. When you call $mqtt->disconnect(). The connection is closed, but remains in the connections array.
When trying to reopen the connection on the second process, it fails. The socket is not open.

You have to call MQTT::disconnect(). This closes the socket and removes connection from the array.

```php
$topic = sprintf("Vennens/MVA/%d/Silo/%d/Alternatieven", $this->plc, $this->silo);
$mqtt = MQTT::connection();
$mqtt->publish($topic, json_encode([
    'alternatief_1' => $this->alternatief_1,
    'alternatief_2' => $this->alternatief_2,
    'alternatief_3' => $this->alternatief_3,
    'alternatief_4' => $this->alternatief_4,
]), 2);

$mqtt->loop(true, true, 5);
MQTT::disconnect();
```

![image](https://user-images.githubusercontent.com/6877485/148641976-270c0780-caee-4101-8bd8-4be0efc658ff.png)

**Comment by Namoshek** [comment]
Thank you for reporting this! I should be able to change the behavior of `MQTT::connection()` to reconnect, if disconnected previously.

In your case, it might be better to not disconnect at all and use an increased timeout to keep the connection alive, at least if you publish messages frequently (at most every 5 minutes). Connecting to the broker is a relatively expensive and time consuming task.

**Comment by stijn1989** [comment]
> I should be able to change the behavior of MQTT::connection() to reconnect, if disconnected previously.
That's also a solution for this problem. Great library, love it.

> Connecting to the broker is a relatively expensive and time consuming task.
I'm using a middleware which checks if the MQTT is available (on certain routes) with connect/disconnect.
The job is executed on user requests. So it's not scheduled.

The backend is part of a system. The MQTT is just there to communicate with a PLC system. In my example, the user has changed some silo's and the PLC needs to know which ones.

**Comment by Namoshek** [comment]
Sounds reasonable. The bug was fixed in #19 and released as [`v1.0.1`](https://github.com/php-mqtt/laravel-client/releases/tag/v1.0.1).
